src/RBTree.mo

```diff
@@ -8,80 +39,286 @@ import O "Order";
 
-    var tree : Tree<X, Y> = (#leaf : Tree<X, Y>);
+    var tree : Tree<K, V> = (#leaf : Tree<K, V>);
 
-    /// Tree as sharable data.
+    /// Return a snapshot of the internal functional tree representation as sharable data.
+    /// The returned tree representation is not affected by subsequent changes of the `RBTree` instance.
     ///
-    /// Get non-OO, purely-functional representation:
-    /// for drawing, pretty-printing and non-OO contexts
-    /// (e.g., async args and results):
-    public func share() : Tree<X, Y> {
+    ///
+    /// Example:
+    /// ```motoko include=initialize
+    ///
+    /// tree.put(1, "one");
+    /// let treeSnapshot = tree.share();
+    /// tree.put(2, "second");
+    /// RBTree.size(treeSnapshot) // => 1 (Only the first insertion is part of the snapshot.)
+    /// ```
+    ///
+    /// Useful for determining the size, pretty-printing, and sharing it across async function calls,
+    /// i.e. passing it in async arguments or async results.
+    ///
+    /// Runtime: `O(1)`.
+    /// Space: `O(1)`.
+    public func share() : Tree<K, V> {
       tree
     };
 
-    /// Get the value associated with a given key.
-    public func get(x : X) : ?Y {
-      getRec(x, compareTo, tree)
+    /// Retrieve the value associated with a given key, if present. Returns `null`, if the key is absent.
+    /// The key is searched according to the `compareTo` function defined on the class instantiation.
+    ///
+    /// Example:
+    /// ```motoko include=initialize
+    ///
+    /// tree.put(1, "one");
+    /// tree.put(2, "two");
+    ///
+    /// tree.get(1) // => ?"one"
+    /// ```
+    ///
+    /// Runtime: `O(log(n))`.
+    /// Space: `O(1)` retained memory plus garbage, see the note below.
+    /// where `n` denotes the number of key-value entries stored in the tree and
+    /// assuming that the `compareTo` function implements an `O(1)` comparison.
+    ///
+    /// Note: Creates `O(log(n))` temporary objects that will be collected as garbage.
+    public func get(key : K) : ?V {
+      getRec(key, compareTo, tree)
     };
 
-    /// Replace the value associated with a given key.
-    public func replace(x : X, y : Y) : ?Y {
-      let (res, t) = insertRoot(x, compareTo, y, tree);
+    /// Replace the value associated with a given key, if the key is present.
+    /// Otherwise, if the key does not yet exist, insert the key-value entry.
+    ///
+    /// Returns the previous value of the key, if the key already existed.
+    /// Otherwise, `null, if the key did not yet exist before.
+    ///
+    /// Example:
+    /// ```motoko include=initialize
+    /// import Iter "mo:base/Iter";
+    ///
+    /// tree.put(1, "old one");
+    /// tree.put(2, "two");
+    ///
+    /// let oldEntry = tree.replace(1, "new one"); // => ?"old one"
+    /// // tree now contains [(1, "new one"), (2, "two")]
+    /// (oldEntry, Iter.toArray(tree.entries())) /// => (?"old one", [(1, "new one"), (2, "two")])
```

Comment:
nit: I think it's okay to not display `oldEntry`. Seems to clutter the example, and I think the important part is that the old entry was overwritten.

Reply:
```
tree.put(1, "old one");
tree.put(2, "two");
ignore tree.replace(1, "new one");
Iter.toArray(tree.entries()) // [(1, "new one"), (2, "two")]
```

Reply:
Indeed. This makes it much clearer. I improved this, same for `remove()`.